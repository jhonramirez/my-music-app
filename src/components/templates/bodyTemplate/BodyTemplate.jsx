import { useState } from "react";
import { Alert, Footer, SongList } from "../..";
import "./bodyTemplate.scss";

export const BodyTemplate = ({ title, songs }) => {
  const [message, setMessage] = useState("");
  const [currentTrack, setCurrentTrack] = useState();
  return (
    <>
      <div className="bodyTemplate">
        <h2 className="bodyTemplate__title">{title}</h2>
        {songs?.map((item, index) => (
          <SongList
            track={item.track}
            favorite={item.favorite}
            key={index}
            setMessage={setMessage}
            onClick={(track) => setCurrentTrack(track)}
          />
        ))}

        <Alert message={message} />
      </div>
      <div className="bodyTemplate__footer">
        <Footer urlTrack={currentTrack?.preview_url} />
      </div>
    </>
  );
};
