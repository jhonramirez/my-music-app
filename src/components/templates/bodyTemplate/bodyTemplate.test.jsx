import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { BodyTemplate } from "./BodyTemplate";

describe("Pruebas en <BodyTemplate/>", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(
      <BodyTemplate title={"proof"} songs={[{ track: {}, favorite: true }]} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Boton Login debe de llamar el getLogin", () => {
    render(
      <BodyTemplate
        title={"proof"}
        songs={[{ track: { preview_url: "url" }, favorite: true }]}
      />
    );
    const BodyTemplateonClick = screen.getByTestId("songList");
    const footer = screen.getByTestId("footer");
    fireEvent.click(BodyTemplateonClick);
    expect(footer).toBeDefined();;
  });
});
