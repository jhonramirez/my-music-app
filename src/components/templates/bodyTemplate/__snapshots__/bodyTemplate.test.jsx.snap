// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <BodyTemplate/> Debe hacer match con el snapshot 1`] = `
<div>
  <section
    class="bodyTemplate"
  >
    <h2
      class="bodyTemplate__title"
    >
      proof
    </h2>
    <article
      class="bodyTemplate__songs"
    >
      <section
        class="songList"
        data-testid="songList"
      >
        <article
          class="songList__img"
        >
          <img
            alt="album"
            class="img"
          />
        </article>
        <article
          class="songList__infoSection"
        >
          <p
            class="songList__info"
          >
            <label
              class="songList__name"
              data-testid="label"
            />
            <label
              class="songList__singer"
              data-testid="label"
            />
          </p>
          <input
            checked=""
            data-testid="heart-input"
            type="checkbox"
          />
          <label
            data-testid="label"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 21.35l-1.45-1.32c-5.15-4.67-8.55-7.75-8.55-11.53 0-3.08 2.42-5.5 5.5-5.5 1.74 0 3.41.81 4.5 2.09 1.09-1.28 2.76-2.09 4.5-2.09 3.08 0 5.5 2.42 5.5 5.5 0 3.78-3.4 6.86-8.55 11.54l-1.45 1.31z"
              />
            </svg>
          </label>
        </article>
      </section>
    </article>
    <div
      id="snackbar"
    />
  </section>
  <section
    class="bodyTemplate__footer"
  >
    <section
      class="footer"
      data-testid="footer"
    >
      <article
        class="footer__soundbar"
      />
      <article>
        <div
          class="footer__player"
          style="width: 640px; height: 360px;"
        />
        <button
          class="btn btn__black"
          data-testid="button"
        >
          <svg
            class=""
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M13 2.5L5 7.119V3H3v10h2V8.881l8 4.619z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="btn btn__black"
          data-testid="button"
        >
          <svg
            class="bi bi-play"
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
             
            <path
              d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"
            />
             
          </svg>
        </button>
        <button
          class="btn btn__black"
          data-testid="button"
        >
          <svg
            class=""
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M11 3v4.119L3 2.5v11l8-4.619V13h2V3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </article>
    </section>
  </section>
</div>
`;
